src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/MappedSubText.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Razor
+{
+    public sealed record MappedSubText(SourceText Text, string FilePath, int StartingLine)
+    {
+        public string SourceLocation => $"{FilePath}@{StartingLine}";
+
+        public static async Task<IEnumerable<MappedSubText>> GetMappedSubTextsAsync(Document document, CancellationToken token)
```

Comment:
I'm fine with this, but I'm still trying to figure out when to return a `Task<IEnumerable<>>` vs an `IAsyncEnumerable<>`. How do you think about that?

Reply:
Yea, that can be a touch decision. In this case, there are two reasons I prefer `Task<IEnumerable>`:

1. The way the enumerable is produced: In this case, all async work is done prior to creating the enumerable items so IAsyncEnumerable wouldn't provide any benefit.
2. The way the enumerable is consumed: The caller in this case is calling the method twice and wants to create lookups from the enumerables. This is a lot simple with IEnumerable than with IAsyncEnumerable.

So, in this scenario, at least, I think `Task<IEnumerable>` is the easy choice since it makes the caller simpler and the actual enumeration part of this API doesn't await anything. If there were awaits while enumerating, I think `IAsyncEnumerable` would make more sense.